Accept a wider range of FHIR resource inputs

apeteri opened the pull request:
Changes are based on representative R4 Core and third-party `CodeSystem` sources.

apeteri committed:
SO-5453: Ignore property "extension" on FHIR Concept representation

apeteri committed:
SO-5451: Add "SUPPLEMENT" as an allowed CodeSystemContentMode value

apeteri committed:
feat(fhir): Add support for "code" extension type

apeteri committed:
feat(fhir): Throw exception if the required service for FHIR "write"...

...operations is not registered

apeteri committed:
fix(fhir): Make DivValidator more lenient; fix self-closing tag

apeteri committed:
fix(fhir): Support all FHIR-approved date formats in dateTime properties

cmark commented on fhir/com.b2international.snowowl.fhir.core/src/com/b2international/snowowl/fhir/core/model/CodeExtension.java:
```diff
@@ -0,0 +1,71 @@
+ * FHIR Code Extension
+ * 
+ * @see <a href="https://www.hl7.org/fhir/extensibility.html#Extension">FHIR:Foundation:Extensibility</a>
+ * @since 8.3.1
```
Use 8.4.0, that's the next minor version we are going to release.

apeteri committed:
fix(fhir): Accept opening div tag without attributes as well

apeteri committed:
fix(fhir.rest.tests): Update FHIR test launch configurations

cmark commented on fhir/com.b2international.snowowl.fhir.core/src/com/b2international/snowowl/fhir/core/request/codesystem/FhirCodeSystemPutRequest.java:
```diff
@@ -42,16 +43,18 @@ public FhirCodeSystemPutRequest(CodeSystem codeSystem) {
+			throw new IllegalStateException("FHIR CodeSystem resource creation support is missing.");
+		}
+		
+		cudSupport.get(0).updateOrCreateCodeSystem(context, codeSystem);
```
I would refactor this to let plug-in developers register their own singleton implementation of `FhirCodeSystemCUDSupport` instead of letting each terminology plugin register their own where only one is being used at runtime.

apeteri committed:
fix(fhir): Change "since" javadoc tag on CodeExtension

apeteri committed:
refactor(fhir.core): Look for a single create-update-delete...

...FHIR CodeSystem service as opposed to inspecting all repositories.
Plugins should register an implementation on the top level
ServiceProvider of Snow Owl.

cmark commented on fhir/com.b2international.snowowl.fhir.core/src/com/b2international/snowowl/fhir/core/request/codesystem/FhirCodeSystemPutRequest.java:
```diff
@@ -43,18 +42,12 @@ public FhirCodeSystemPutRequest(CodeSystem codeSystem) {
-		
+		final Optional<FhirCodeSystemCUDSupport> cudSupport = context.optionalService(FhirCodeSystemCUDSupport.class);
 		if (cudSupport.isEmpty()) {
 			throw new IllegalStateException("FHIR CodeSystem resource creation support is missing.");
```
How about throwing an `HTTP 501 Not Implemented` in this case?

cmark replied:
@apeteri did you apply this requested change? If not, could you please do so, thanks!

cmark replied:
Will apply this change directly to 8.x after merge.

cmark replied:
Done
